lib/index.js

```diff
@@ -257,67 +257,71 @@ exports.setSchema = function (schema) {
 
         foundObjects.get(type).set(JSON.stringify(obj.id), obj);
 
-        // fetch all relations
+        var async = idOrIds && idOrIds.async;
         var subTasks = [];
-        Object.keys(typeInfo.relations || {}).forEach(function (field) {
-          var relationDef = typeInfo.relations[field];
-          var relationType = Object.keys(relationDef)[0];
-          var relatedType = relationDef[relationType];
-          if (typeof relatedType !== 'string') {
-            var relationOptions = relatedType.options;
-            if (relationOptions && relationOptions.async) {
-              return;
-            }
-            relatedType = relatedType.type;
-          }
-          if (relationType === 'belongsTo') {
-            var relatedId = obj[field];
-            if (typeof relatedId !== 'undefined') {
-              subTasks.push(Promise.resolve().then(function () {
-
-                // short-circuit if it's already in the foundObjects
-                // else we could get caught in an infinite loop
-                if (foundObjects.has(relatedType) &&
-                    foundObjects.get(relatedType).has(JSON.stringify(relatedId))) {
-                  return;
-                }
-
-                // signal that we need to fetch it
-                return {
-                  relatedType: relatedType,
-                  relatedIds: [relatedId]
-                };
-              }));
+
+        if (!async) {
+          // fetch all relations
+          Object.keys(typeInfo.relations || {}).forEach(function (field) {
+            var relationDef = typeInfo.relations[field];
+            var relationType = Object.keys(relationDef)[0];
+            var relatedType = relationDef[relationType];
+            if (typeof relatedType !== 'string') {
+              var relationOptions = relatedType.options;
+              if (relationOptions && relationOptions.async && (async === undefined)) {
```

Comment:
Also the reason it's better to do `typeof foo === 'undefined'` rather than `foo === undefined` is because technically people can re-assign the `undefined` object in JavaScript